src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,794 @@
+        pe = self.get_nodeattr("PE")
+        return num_channels // pe
+
+    def infer_node_datatype(self, model):
+        """Used for FINN DataType inference: set the output tensors' datatypes
+        accordingly for this node"""
+        node = self.onnx_node
+        idt = model.get_tensor_datatype(node.input[0])
+        if idt != self.get_input_datatype():
+            warn_str = "inputDataType changing for %s: %s -> %s " % (
+                node.name,
+                str(self.get_input_datatype().name),
+                str(idt.name),
+            )
+            warnings.warn(warn_str)
+        self.set_nodeattr("inputDataType", idt.name)
+        # set output datatype from property
+        odt = self.get_output_datatype()
+        model.set_tensor_datatype(node.output[0], odt)
+
+    def verify_node(self):
+        """Required by the FINN nalysis module. Checks if custom ops in graph
+        are correctly built, with all attributes and inputs."""
+        return []
+
+    def bram_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("BRAM", 0)
+
+    def uram_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("URAM", 0)
+
+    def lut_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("LUTRAM", 0)
+
+    def get_input_datatype(self, ind=0):
+        return DataType[self.get_nodeattr("inputDataType")]
+
+    def get_output_datatype(self, ind=0):
+        return DataType[self.get_nodeattr("outputDataType")]
+
+    def get_weight_datatype(self):
+        """The term 'weights' and 'thresholds' are used interchangably in this class."""
+        return DataType[self.get_nodeattr("weightDataType")]
+
+    def minimize_accumulator_width(self, model):
+        "Minimize threshold width ('accumulator width' here due to convention)"
+        thresholds = model.get_initializer(self.onnx_node.input[1])
+        threshold_tensor = self.get_hw_compatible_threshold_tensor(thresholds)
+        min_threshold = thresholds.min()
+        max_threshold = thresholds.max()
+        min_input = self.get_input_datatype().min()
+        max_input = self.get_input_datatype().max()
+        # get range required by threshold values
+        tdt_min = min(min_input, min_threshold)
+        tdt_max = max(max_input, max_threshold)
+        if tdt_min < 0:
+            if abs(tdt_min) > tdt_max:
+                tdt = DataType.get_smallest_possible(tdt_min)
+            else:
+                tdt = DataType.get_smallest_possible(-tdt_max - 1)
+        else:
+            tdt = DataType.get_smallest_possible(tdt_max)
+        assert np.vectorize(tdt.allowed)(
+            threshold_tensor
+        ).all(), "Thresholds can't be expressed with type %s" % str(tdt)
+        self.set_nodeattr("weightDataType", tdt.name)
+        return DataType[self.get_nodeattr("weightDataType")]
+
+    def get_instream_width(self, ind=0):
```

Comment:
All of the functions below are already defined in Thresholding. By letting Thresholding_rtl directly inherit from Thresholding, there is no need to redefine these functions and if they are identical between the HLS and RTL variant, they can also be moved to the HW op and deleted from `Thresholding_rtl` & `Thresholding_hls`, like `get_weightstream_width`.